internal/ingress/controller/controller.go

```diff
@@ -231,12 +235,21 @@ type KongController struct {
 	PluginSchemaStore PluginSchemaStore
 
 	Logger logrus.FieldLogger
+
+	tmpDir string
```

Comment:
This is invalid if `DumpConfig` is not enabled. Therefore, the name of this field should be closely tied to that functionality, and there should be a comment saying in which circumstances this field is usable.

Something along the lines of

```go
// dumpConfigDir is the target directory for config dumps, if dumping is enabled.
dumpConfigDir string
```